src/main/java/teammates/common/datatransfer/attributes/StudentProfileAttributes.java

```diff
@@ -245,4 +245,128 @@ private boolean isGenderValid(String gender) {
+        private UpdateOption<String> genderOption = UpdateOption.empty();
+        private UpdateOption<String> moreInfoOption = UpdateOption.empty();
+        private UpdateOption<String> pictureKeyOption = UpdateOption.empty();
+
+        private UpdateOptions(String googleId) {
+            Assumption.assertNotNull(Const.StatusCodes.UPDATE_OPTIONS_NULL_INPUT, googleId);
+
+            this.googleId = googleId;
+        }
+
+        public String getGoogleId() {
+            return googleId;
+        }
+
+        /**
+         * Builder class to build {@link UpdateOptions}.
+         */
+        public static class Builder {
+            private UpdateOptions updateOptions;
+
+            private Builder(String googleId) {
+                updateOptions = new UpdateOptions(googleId);
+            }
+
+            public Builder withShortName(String shortName) {
+                Assumption.assertNotNull(Const.StatusCodes.UPDATE_OPTIONS_NULL_INPUT, shortName);
+
+                updateOptions.shortNameOption = UpdateOption.of(shortName);
+                return this;
+            }
+
+            public Builder withEmail(String email) {
+                Assumption.assertNotNull(Const.StatusCodes.UPDATE_OPTIONS_NULL_INPUT, email);
+
+                updateOptions.emailOption = UpdateOption.of(email);
+                return this;
+            }
+
+            public Builder withInstitute(String institute) {
+                Assumption.assertNotNull(Const.StatusCodes.UPDATE_OPTIONS_NULL_INPUT, institute);
+
+                updateOptions.instituteOption = UpdateOption.of(institute);
+                return this;
+            }
+
+            public Builder withNationality(String nationality) {
+                Assumption.assertNotNull(Const.StatusCodes.UPDATE_OPTIONS_NULL_INPUT, nationality);
+
+                updateOptions.nationalityOption = UpdateOption.of(nationality);
+                return this;
+            }
+
+            public Builder withGender(String gender) {
+                Assumption.assertNotNull(Const.StatusCodes.UPDATE_OPTIONS_NULL_INPUT, gender);
+
+                updateOptions.genderOption = UpdateOption.of(gender);
+                return this;
+            }
+
+            public Builder withMoreInfo(String moreInfo) {
+                Assumption.assertNotNull(Const.StatusCodes.UPDATE_OPTIONS_NULL_INPUT, moreInfo);
+
+                updateOptions.moreInfoOption = UpdateOption.of(moreInfo);
+                return this;
+            }
+
+            public Builder withPictureKey(String pictureKey) {
+                Assumption.assertNotNull(Const.StatusCodes.UPDATE_OPTIONS_NULL_INPUT, pictureKey);
+
+                updateOptions.pictureKeyOption = UpdateOption.of(pictureKey);
+                return this;
+            }
+
+            public UpdateOptions build() {
+                return updateOptions;
+            }
+
+        }
+
+        @Override
+        public String toString() {
```

Comment:
Do we have a use for this `toString()`?

Reply:
For logging purpose in order to show what is being updated.

The benefits are not showing in this commit.